matrix/plugins/glitch/main.py

```diff
@@ -73,25 +69,30 @@ def select(model, selectors, objects=None, resolver=default_resolver):
             model,
             num=action.args.get('glitch_num', 5)))
 
+    rule.log.info("Writing glitch plan to {}".format(output_filename))
+    with open(output_filename, 'w') as output_file:
+        output_file.write(yaml.dump(glitch_plan))
+
     # Execute glitch plan. We perform destructive operations here!
     for action in glitch_plan['actions']:
-        actionf = Actions[action.pop('action')]
+        actionf = Actions[action.pop('action')]['func']
         selectors = action.pop('selectors')
         # Find a set of units to act upon
         objects = select(model, selectors)
 
         # Run the specified action on those units
         rule.log.debug("GLITCHING {}: {}".format(actionf.__name__, action))
+
+        # TODO: better handle the case where we no longer have objects
+        # to select, due to too many of them being destroyed (most
+        # relevant for small bundles)
+        await actionf(rule, model, objects, **action)
         context.bus.dispatch(
             origin="glitch",
-            payload=functools.partial(actionf, model, objects, **action),
+            payload={'action': actionf.__name__, **action},
```

Comment:
we might be better off here if this were the inspect.Signature object as well. It should handle things like the __qualname__ vs __name__ handling